chains/solana/contracts/crates/build-commit/src/lib.rs

```diff
@@ -0,0 +1,67 @@
+use std::process::Command;
+
+pub fn cargo_instructions(source_file: &str) {
+    let program_name = source_file
+        .split('/')
+        .nth_back(1)
+        .unwrap()
+        .trim()
+        .to_string();
```

Comment:
I think using `Path` would be a bit cleaner here, but feel free to ignore.
```
Path::from(source_file).parent().file_name();
```

(paraphrasing from memory, you might need some unwraps in there.